jacodb-analysis/src/main/kotlin/org/jacodb/analysis/config/TaintAction.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ *  Copyright 2022 UnitTestBot contributors (utbot.org)
+ * <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ * <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.jacodb.analysis.config
+
+import org.jacodb.analysis.ifds2.taint.Tainted
+import org.jacodb.analysis.paths.AccessPath
+import org.jacodb.taint.configuration.Action
+import org.jacodb.taint.configuration.AssignMark
+import org.jacodb.taint.configuration.CopyAllMarks
+import org.jacodb.taint.configuration.CopyMark
+import org.jacodb.taint.configuration.PositionResolver
+import org.jacodb.taint.configuration.RemoveAllMarks
+import org.jacodb.taint.configuration.RemoveMark
+import org.jacodb.taint.configuration.TaintActionVisitor
+
+class TaintActionEvaluator(
+    internal val positionResolver: PositionResolver<AccessPath>,
+) {
```

Comment:
It is not possible here, because TaintActionEvaluator is NOT aware of a fact, only FactAwareTaintActionEvaluator is. So I made them both NOT visitors, but simple classes.

Reply:
Let me rephrase: `TaintActionEvaluator` cannot implement TaintActionVisitor, because 4 of 5 actions require "current fact" in order to be "evaluated". This "fact" could be made a constructor parameter -- in this case we can impl ActionVisitor -- this is exactly what `FactAwareTaintActionEvaluator` is. On the other hand, `TaintActionEvaluator` serves a different purpose: provide a way to be able to evaluate some actions WITHOUT knowing the current fact -- this is useful in cases when we do not know "the current fact" and either do not care about actions that require it or can guarantee that we will not meet them.

Reply:
Do we really need `FactAwareTaintActionEvaluator` then?

Reply:
Currently, it is unused, but it is useful as a reference implementation of self-contained Visitor-based taint action evaluator.